frontend/src/types/apiTypes.ts

```diff
@@ -0,0 +1,29 @@
+export class ApiError extends Error {
+  status: string;
+  body: any;
+
+  constructor(res: any, body: any) {
+    super(body.message || res.statusText);
+    this.status = res.status;
+    this.name = 'ApiError';
+    this.body = body;
+  }
+}
+
+export type ErrorState = {
+  apiError: ApiError | null;
+};
+
+export type SetErrorStateAction = {
+  type: typeof SET_ERROR;
+  payload: ApiError;
```

Comment:
If this is meant to be the way to display all errors we should probably have just Error be the payload type. Can just rename the file to errorTypes.ts as well.